### ISSUE TITLE
Sudo

### ISSUE THREAD

**Issue by Grimmy347** [onebeartoe/electronics#2]
how can I get it to run all the commands as sudo?  With the most recent version of raspbian it does not want to access the memory due to user permissions.

**Comment by onebeartoe** [comment]
Which commands, sir?

**Comment by Grimmy347** [comment]
so I am trying to use your webapp to run the led-matrix that was on adafruit.  I didn't want to use payara so I have installed apache and tomcat8 and I put the .war into the tomcat root folder and it works almost the same.  I had to create the .onebeartoe folder for the settings in the tomcat folder but other then that, it all looked like it was working.  The only issue is nothing is showing up on my matrix.  I can run the command from the matrix folder and it works fine.  I looked at tomcats logs and the command it is executing looks like it should work except it is not typing sudo before the command.  I tried doing the same, running the command without sudo first and it gave me an error about not being able to access /dev/mem,  If I could get the applet to add sudo before the command then it looks like it would work.  I tried to add it into the command in java and recompiling but I have not been able to get that to work mostly because I don't know how the ProcessBuilder works.  Any ideas how I could force it to use that command?

**Comment by Grimmy347** [comment]
Exact response when I try to run the command without doing sudo is "Must run as root to be able to access /dev/mem.  Prepend 'sudo' to the command"

**Comment by Grimmy347** [comment]
I tried to trick it to make it work by adding a command line flag of "& sudo entirecommandagain" and what printed out in tomcat logs looks like it should have worked but still nothing is printing out on the matrix

**Comment by onebeartoe** [comment]
Thanks for the details, Grimmy347.  

Firstly, that project only officially suppports Payara for the application container.  So I recomend that you follow the instructions on the Adafruit guide, since those are known to work.

But I am familiar with Tomcat and the issue for sure sounds like a permission issue on your target system.

From what I remember and re-researched the officially supported intructions do not require root level premissons.  So it is hard to tell what is going on with your setup.

Have you tried running the Tomcat start command with sudo?

**Comment by Grimmy347** [comment]
I did and eventually I was able to get it to run the command by inserting "sudo" into the proccess builder before everything else but it didn't want to end the process once it started and was using up a ton of cpu.  I could not get it to work so I have given up and I am back with payara which I think was probably the better choice but I wanted to try something different.  Thanks for your help.  Your work is the core of my project and is what is allowing me to get my vision working!
